packages/backend/src/email/base-mail.ts

```diff
@@ -0,0 +1,51 @@
+import type { EmailOptions } from '@types';
+import { getEnv } from '@helpers';
+import type { EmailService } from '../services/email/email';
+import { emailService } from '../services/email';
+import type { MailInterface } from './mail-interface';
+
+export class BaseMail implements MailInterface {
```

Comment:
MailInterface -> IMailBuilder or MailBuilder

Reply:
and BasMale can be changed as well

Reply:
Maybe we shouldn't rename such classes, because of course it seems that it is builder, but we the main thing here is that this is Mail entity, so we don't care what pattern that entity implements.

Reply:
It's not about the name of the pattern, it's instead what it does. The same as EmailSender